### ISSUE TITLE
Shader issues on Intel UHD Graphics 620

### ISSUE THREAD

**Issue by leso-kn** [3dreamengine/3DreamEngine#55]
Hi,

I've recently discovered this project and first of all, amazing work!

Now, unfortunately on my main machine, which is an Intel NUC Kit 7i7, there seems to be a major shader issue.

To describe the problem in more detail: it seems like the color space is completely shifted or like the buffers are mixed up or something simliar. I have attached screenshots of the tavern, alpha blend and fps game examples.

**I'm running**

* Ubuntu 20.10 64bit
* `love-11.3`
* Intel UHD Graphics 620
* 16 GB of RAM
* `libegl-mesa0` version `20.2.6`

**Things I've tried so far**

On the tavern demo, I was as well experiencing the black screen issue with smooth lighting turned on as described in #33, without that it looks like in the screenshot.

I have also tried most fixes mentioned in #33. What actually surpises me, is that @Luke100000 uses _precisely_ this GPU to validate shaders. So something must be very fishy with my environment.

I wonder if this is a known issue, didn't find anything similar among the Github issues.
Also, please let me know if you need more specifications on my environment.

**Screenshots**
<img alt="alpha blend 2021-02-23 21-30-44" src=https://user-images.githubusercontent.com/7886606/108905985-a8536200-7620-11eb-84b5-9fce220457d6.png width=33% /><img alt="tavern 2021-02-23 21-32-00" src=https://user-images.githubusercontent.com/7886606/108905997-abe6e900-7620-11eb-9d46-c91cb429fecc.png width=33% /><img alt="fps game 2021-02-23 21-32-53" src=https://user-images.githubusercontent.com/7886606/108906009-aee1d980-7620-11eb-8d35-e52cd19a8e82.png width=33% />

**Comment by Luke100000** [comment]
Thanks for the detailed issue. That looks like modern art!

I was able to reconstruct the environment using a live version of ubuntu 20.04 on my UHD 620 system. While the result is slightly different it probably is the same error.

After some try and error I found the error, and I am a little bit ashamed. `vec3 col;` does NOT initialize with 0 on all systems...

Thanks again, I will double check for similar errors and then fix it on both master and beta branch.
